### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Koochr**
What was done:
- Promisified some IPC calls
- Refactored all fs calls (except `existsSync`) to be async. Also changed all `fs` calls to `fs-extra` and replaced `rimraf` with `rmrf`, which is its more modern and async version
- Fixed a bug with one word appearing twice in verifying keywords
- Fixed various bugs and ended up completely rewriting the windows creating system (see below)

Problems with window starting system I fixed:
- We were facing race conditions. I could only uncover it after fs refactoring to be async: since it became faster, we started to running into the situation, when `ready` event was already fired at the moment when we tried to create dashboard window, and window never get created
- Most of electron event listeners were redundant
- We were facing race condition in another situation: when closing one window and opening another (e.g. after install or on logout). If the first window was closed before the second one is opened, the app was closed. I also think we mostly evaded this issue bc of plenty of sync fs calls, which was slowing down the process.
- After I fixed the above one, i found another issue: every window was calling `ipcMain.removeAllListeners`, while some listeners were shared. It led, for instance, to the situation, when welcome window could not be closed after logout. To fix it, I changed the way of removing listeners to the loop through existing events.

### CODE REVIEW

File: shared/helpers.ts [hunk 1]
```diff
@@ -66,7 +65,7 @@ const getInstalledVersionInfo: (resource: 'node' | 'firefox' | 'sdk') => {
   const pointPath = getPointPath()
   try {
     return JSON.parse(
-      fs.readFileSync(path.join(pointPath, `${file}.json`)).toString()
+      (await fs.readFile(path.join(pointPath, `${file}.json`))).toString()
```
Comment (JatinBumbra) [inline shared/helpers.ts:68]: I'm actually pretty confused. We're making functions `async` from sync and yet we're `await`ing them, which yields sync behavior only. What's the point then?

**Review by JatinBumbra — CHANGES_REQUESTED** [review]

File: shared/helpers.ts (hunk 1 above)
Comment (Koochr) [reply shared/helpers.ts:68]: `await` doesn't convert async function to sync, it's just syntactic sugar and is equivalent to `then()`.
And it's a huge difference, sync calls block the main thread, while async don't. Actually, you should always avoid sync calls to fs, it makes the whole app much slower.

Comment (JatinBumbra) [reply shared/helpers.ts:68]: I read a couple of threads on Stack Overflow and totally agree with you here. Sync calls block the entire thread, so any other tasks running too will stop, while `await` only waits for the `await`ed function to return before proceeding further, while other tasks can continue running. PR approved from my side.

**Review by JatinBumbra — APPROVED** [review]

**Comment by JatinBumbra** [comment]
@Koochr There are some merge conflicts. Can you resolve them?

**Pull request merged**